csharp/ExcelAddInInstaller/CustomActions/RegistryManager.cs

```diff
@@ -0,0 +1,234 @@
+﻿using Microsoft.Win32;
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace Deephaven.ExcelAddInInstaller.CustomActions {
+  public class RegistryManager {
+    public static bool TryCreate(Action<string> logger, out RegistryManager result, out string failureReason) {
+      result = null;
+      failureReason = "";
+
+      var regView = Environment.Is64BitOperatingSystem ? RegistryView.Registry64 : RegistryView.Registry32;
+      var regBase = RegistryKey.OpenBaseKey(RegistryHive.LocalMachine, regView);
+
+      if (!TryOpenSubKey(regBase, RegistryKeys.Bitness.Key, false,
+            out var bitnessKey, out failureReason) ||
+          !TryOpenSubKey(Registry.CurrentUser, RegistryKeys.OpenEntries.Key, true,
+            out var openKey, out failureReason)) {
+        return false;
+      }
+
+      result = new RegistryManager(bitnessKey, openKey, logger);
+      return true;
+    }
+
+    private static bool TryOpenSubKey(RegistryKey baseKey, string key, bool writable, out RegistryKey result, out string failureReason) {
+      failureReason = "";
+      result = baseKey.OpenSubKey(key, writable);
+      if (result == null) {
+        failureReason = $"Couldn't find registry key {RegistryKeys.Bitness.Key}";
+        return false;
+      }
+
+      return true;
+    }
+
+
+    public static bool TryMakeAddInEntryFromPath(string path, out string result, out string failureReason) {
+      result = "";
+      failureReason = "";
+      if (path.Contains("\"")) {
+        failureReason = "Path contains illegal characters";
+        return false;
+      }
+      result = $"/R \"{path}\"";
+      return true;
+    }
+
+    private readonly RegistryKey _registryKeyForBitness;
+    private readonly RegistryKey _registryKeyForOpen;
+    private readonly Action<string> _logger;
+
+    public RegistryManager(RegistryKey registryKeyForBitness, RegistryKey registryKeyForOpen,
+      Action<string> logger) {
+      _registryKeyForBitness = registryKeyForBitness;
+      _registryKeyForOpen = registryKeyForOpen;
+      _logger = logger;
+    }
+
+    /// <summary>
+    /// Determine whether the installed version of Office is the 32-bit version or the 64-bit version.
+    /// It's possible for example that the 32-bit version of Office can be installed on a 64-bit OS.
+    /// </summary>
+    /// <param name="is64Bit"></param>
+    /// <param name="failureReason"></param>
+    /// <returns></returns>
+

```

Comment:
Is this space between the comment block and the function signature intended?

Reply:
Deleted, thanks